ntbs-service/DataAccess/DataQualityRepository.cs

```diff
@@ -102,6 +104,39 @@ public async Task<IList<Notification>> GetNotificationsEligibleForDotVotAlerts()
                 .ToListAsync();
         }
 
+        public async Task<IList<NotificationAndDuplicateIds>> GetNotificationIdsEligibleForPotentialDuplicateAlerts()
+        {
+            return await GetBaseNotificationQueryableForAlerts()
+                .SelectMany(a => _context.Notification, 
+                    (notification, duplicate) => new {notification, duplicate})
+                .Where(t =>
+                    (t.notification.PatientDetails.GivenName == t.duplicate.PatientDetails.FamilyName &&
+                     t.notification.PatientDetails.FamilyName == t.duplicate.PatientDetails.GivenName &&
+                     t.notification.PatientDetails.NhsNumber == t.duplicate.PatientDetails.NhsNumber &&
+                     t.notification.PatientDetails.NhsNumber != null &&
+                     t.notification.PatientDetails.GivenName != null &&
+                     t.notification.PatientDetails.GivenName != null) 
+                    ||
+                    (t.notification.PatientDetails.NhsNumber == t.duplicate.PatientDetails.NhsNumber &&
+                     t.notification.PatientDetails.Dob == t.duplicate.PatientDetails.Dob && 
+                     t.notification.PatientDetails.NhsNumber != null &&
+                     t.notification.PatientDetails.Dob != null)
+                    &&
+                    t.notification.NotificationDate < DateTime.Now.AddDays(-DataQualityPotentialDuplicateAlert.MinNumberDaysNotifiedForAlert) &&
+                    t.duplicate.NotificationDate < DateTime.Now.AddDays(-DataQualityPotentialDuplicateAlert.MinNumberDaysNotifiedForAlert))
+                .Select(t => new NotificationAndDuplicateIds
+                {
+                    NotificationId = t.notification.NotificationId,
+                    DuplicateId = t.duplicate.NotificationId
+                }).ToListAsync();
+        }
```

Comment:
I take form this that the `Join` didn't work for some reason?
Have you had a chance to see what sql this produces? I'm a little worried that `SelectMany` will break the abstraction here and we'll do lots in memory?
If that is the case, and `join` won't help, we can filter the sequences individually (by `NotificationDate`) before applying the select many and pairwise filtering!